### ISSUE TITLE
Failure of all tests due to ::gentoo missing.

### ISSUE THREAD

**Issue by LinoBigatti** [alunduil/etest#79]
All tests are currently failing on the build phase, due to portage not finding ``/var/db/repos/gentoo``. I can think of 2 ways to fix this:

- Run ``emerge-webrsync`` or ``emerge --sync`` at the beginning of each test. This would make the tests run more slowly.
- Bind mount the repo from the host system. This would require scouring the env for the correct location and a gentoo host.

@alunduil let me know your thoughts on the matter.

[Related logs](https://gist.github.com/LinoBigatti/54d32aa68ae18ed5ff9a6aeab41cf146)

**Comment by alunduil** [comment]
I recall an option to specify that as a bind mount using an option.  If not, what about this, we create an option for --gentoo-path and --sync then the user decides if it does a sync internally or bind mounts from the host.  Does that sound reasonable?  We could do two pull requests one for each option if that makes sense too.

**Comment by LinoBigatti** [comment]
Such option doesnt exist, or at least isnt mentioned in ``etest --help``. I thought of a 3rd solution: If the host system is not gentoo, download a repo snapshot to .cache and bind mount that. Do you think that is reasonable?. Also, how would 2 different PRs work?

**Comment by alunduil** [comment]
Yeah, that sounds good to me.  Snapshot in cache would be a good solution indeed.

Just submit two different pull requests for two different changes (I was suggesting one pull request for --gentoo and another for --sync).  Your idea circumvents the need for that so I say go for it.

**Comment by LinoBigatti** [comment]
Ok, Ill do a quick hard-coded draft and if that works Ill add the new functionality.
